### PR TITLE
Created AppBadge component

### LINKED ISSUE #59: Create AppBadge component

### PR DESCRIPTION
**Pull request by D1nviz**
- Added AppBadge component
- Wrote tests

image

### CODE REVIEW

File: code/src/components/app-badge/AppBadge.test.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+describe("AppBadge", () => {
+  it("renders AppBadge component", () => {
+    render(<AppBadge>children</AppBadge>);
+    const child = screen.getByText("children");
+    expect(child).toBeInTheDocument();
+  });
+
+  it("applies the no-round class when isRounded is false", () => {
+    const { container } = render(<AppBadge isRounded={false} />);
+    expect(container.firstChild).toHaveClass("spa-badge__no-round");
```
Comment (mhevyk) [inline code/src/components/app-badge/AppBadge.test.tsx]: Do we need to destructure `container` here? Maybe we can use just `screen.findByText`?

File: code/src/components/app-badge/AppBadge.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import "@/components/app-badge/AppBadge.scss";
+
+const AppBadge = ({
+  className,
+  isRounded = true,
+  size = "medium",
+  color = "contained",
+  ...props
+}: AppBadgeProps) => {
+  const roundVariant = isRounded ? "round" : "no-round";
```
Comment (mhevyk) [inline code/src/components/app-badge/AppBadge.tsx]: Add empty line below

File: code/src/components/app-badge/AppBadge.test.tsx (hunk 1 above)
Comment (D1nviz) [reply code/src/components/app-badge/AppBadge.test.tsx]: fixed

File: code/src/components/app-badge/AppBadge.tsx (hunk 2 above)
Comment (D1nviz) [reply code/src/components/app-badge/AppBadge.tsx]: fixed

File: code/src/components/app-badge/AppBadge.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import "@/components/app-badge/AppBadge.scss";
+
+const AppBadge = ({
+  className,
+  isRounded = true,
+  size = "medium",
+  color = "contained",
+  ...props
+}: AppBadgeProps) => {
+  const roundVariant = isRounded ? "" : "spa-badge__no-rounded";
```
Comment (mhevyk) [inline code/src/components/app-badge/AppBadge.tsx]: What to you think about creating some class concat utility, that can join classes correctly across the project? I was inspired by `tailwind` approaches. For example like following:

```jsx
export function cc(...classNames: unknown[]) {
  return classes.filter(className => typeof className === "string").join(" ");
}
```

Example usage:

```jsx
className={cc(
    "spa-badge",
     variant && `spa-badge__${variant}`
     isRounded ? "" : "spa-badge__no-rounded",
     className
)}
```

I am not sure about naming, maybe you can propose better name.

You should place this function into global `utils` folder

**UPD:** maybe you should also filter out empty strings also to make joined classes look pretty)

Comment (D1nviz) [reply code/src/components/app-badge/AppBadge.tsx]: Great idea! Created the utility function and replaced it in AppBadge component

**Review by mhevyk — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

File: code/src/components/app-badge/AppBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import Badge from "@mui/material/Badge";
+
+import { AppBadgeProps } from "@/components/app-badge/AppBadge.types";
+
+import cn from "@/utils/cn";
+
+import "@/components/app-badge/AppBadge.scss";
```
Comment (dmtrth25) [inline code/src/components/app-badge/AppBadge.tsx]: not sure why do you need so many spaces between imported here, maybe can add spaces only for styles below and that's it like for our previous some code examples when we use space only for style import.
It's not urgent but still for future approaches nice to have

File: code/src/components/app-badge/AppBadge.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+const AppBadge = ({
+  className,
+  isRounded = true,
+  size = "medium",
+  color = "contained",
+  ...props
+}: AppBadgeProps) => {
+  return (
+    <Badge
+      max={10}
```
Comment (dmtrth25) [inline code/src/components/app-badge/AppBadge.tsx]: why are you decided used `max` with `10` value here ?

Comment (D1nviz) [reply code/src/components/app-badge/AppBadge.tsx]: I deleted it. And if we need this max value, we can pass it like props

File: code/src/components/app-badge/AppBadge.tsx (hunk 4 above)
Comment (D1nviz) [reply code/src/components/app-badge/AppBadge.tsx]: fixed

**Review by OliaHrytsak — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**